enterprise/backend/src/metabase_enterprise/sso/integrations/sso_settings.clj

```diff
@@ -91,6 +86,22 @@ on your IdP, this usually looks something like http://www.example.com/141xkex604
   :default {}
   :setter (comp (partial setting/set-value-of-type! :json :saml-group-mappings) validate-group-mappings))
 
+(defsetting saml-enabled
+  (deferred-tru "Enable SAML authentication.")
+  :type    :boolean
+  :default false
+  :setter (fn [new-value]
+            (if new-value
+              ;; Check that require settings are set and valid before allowing SAML to be enabled
+              (if (and (saml-identity-provider-uri)
+                       (saml-identity-provider-certificate))
+                (do
+                 (validate-saml-idp-cert (saml-identity-provider-certificate))
+                 (setting/set-value-of-type! :boolean :saml-enabled new-value))
+                (throw (ex-info (tru "SAML is not configured. Please set the IdP URI and Certificate first.")
+                                {:samle-enabled new-value})))
```

Comment:
Is `:samle-enabled` a typo? Then, feel free to ignore this one, but: a `cond` might work better than the nested ifs here